File: adventofcode-2016/src/main/java/nl/mvdr/adventofcode/adventofcode2016/day15/Part2.java
package nl.mvdr.adventofcode.adventofcode2016.day15;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.Solver;

public class Part2 implements Solver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part2.class);

    @Override
    public String solve(String inputfile) {
        return new TimingIsEverythingSolver(true).solve(inputfile);
    }

    public static void main(String[] args) {
        Part2 instance = new Part2();

        String result = instance.solve("input-day15.txt");

        LOGGER.info(result);
    }
}

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day01/Part2.kt
package nl.mvdr.adventofcode.adventofcode2025.day01

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.point.Direction
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart2(lines: Sequence<String>): Int {
    var dial = 50
    var zeroes = 0
    for (instruction in lines) {
        var distance = instruction.substring(1).toInt()
        val direction = when(Direction.parse(instruction.take(1))) {
            Direction.LEFT -> -1
            Direction.RIGHT -> 1
            else -> throw IllegalArgumentException("Unexpected direction: {direction}")
        }

        // Let's just brute force it.
        // I'm sure this can be done by cleverly using div and mod, but the edge cases seem tricky.
        while (0 < distance) {
            dial = (dial + direction) % 100
            if (dial == 0) {
                zeroes++
            }
            distance--
        }
    }

    return zeroes
}

fun main() {
    val result = FunctionSolver(::solvePart2).solve("input-day01.txt")
    logger.info { result }
}
